ui/pages/confirmations/components/confirm/row/dataTree.tsx

```diff
@@ -94,11 +94,14 @@ export const DataTree = ({
   >(undefined);
 
   useEffect(() => {
-    getTokenDecimalsOfDataTree(data).then((decimals) => {
+    const getDecimals = async () => {
+      const decimals = await getTokenDecimalsOfDataTree(data);
       if (typeof decimals === 'number') {
         setTokenContractDecimals(decimals);
       }
-    });
+    };
+
+    getDecimals();
```

Comment:
Since PR is refactoring this useeffect, it may be useful to use this hook I think to avoid race conditions:

https://github.com/MetaMask/metamask-extension/blob/develop/ui/hooks/useAsyncResult.ts

Reply:
conversation continued below https://github.com/MetaMask/metamask-extension/pull/26324#discussion_r1702107390